api/test/integration/tavern_utils.py

```diff
@@ -328,3 +334,110 @@ def test_validate_search(response, search_param):
 
 def test_validate_key_not_in_response(response, key):
     assert all(key not in item for item in response.json()["data"]["affected_items"])
+
+
+def check_agentd_started(response, agents_list):
+    """Wait until all the agents had their agentd process started correctly. This will avoid race conditions caused by
+    agents reconnections before restarting.
+
+    Parameters
+    ----------
+    response : Request response
+    agents_list : list
+        List of expected agents to be restarted.
+    """
+
+    def get_timestamp(log):
+        """Get timestamp from log.
+
+        Parameters
+        ----------
+        log : str
+            String representing the log to get the timestamp from.
+
+        Returns
+        -------
+        datetime
+            Datetime object representing the timestamp got.
+        """
+        timestamp = re.search(r'^\d\d\d\d/\d\d/\d\d\s\d\d:\d\d:\d\d', log).group(0)
+        return datetime.strptime(timestamp, "%Y/%m/%d %H:%M:%S")
+
+    # Save the time when the restart command was sent
+    restart_request_time = datetime.utcnow().replace(microsecond=0) - response.elapsed
+
+    for agent_id in agents_list:
+        tries = 0
+        while tries < 80:
+            try:
+                # Save agentd logs in a list
+                command = f"docker exec env_wazuh-agent{int(agent_id)}_1 grep agentd /var/ossec/logs/ossec.log"
+                output = subprocess.check_output(command.split()).decode().strip().split('\n')
+            except subprocess.CalledProcessError:
+                assert False, f"Error while trying to get logs from agent {agent_id}"
```

Comment:
This could be `raise AssertionError(f"Error while trying to get logs from agent {agent_id}")`, but we could use a more concrete exception.

Reply:
Used concrete exceptions.